scripts/templates/helper.py

```diff
@@ -266,6 +266,18 @@ def get_struct_members(type_name, meta):
             raise Exception(
                 f"Cannot return members of non-struct type {struct_type}")
         return meta['struct'][struct_type]['members']
+    
+    @staticmethod
+    def is_stype_struct(type_name, specs):
+        # The stype members added from base structs aren't present in the meta
+        # dict, so we need to search through the specs instead.
+        for s in specs:
```

Comment:
Normally I wouldn't be worried about perf of these scripts, but this is adding a loop over the entire spec, nested inside of a few other loops (over every function, over every argument).
Maybe it would be better to add it into the meta dict or create a separate cache?

Did you observe any meaningful difference in the time the generation time?

Reply:
Eyeballing it I didn't notice a change. I've just done some proper testing with `time` and generation time hovers around 25.9 +/- 0.15 seconds on my machine both before and after this commit